app/src/main/java/com/codepath/apps/critterfinder/fragments/PetFavoritesFragment.java

```diff
@@ -59,6 +63,12 @@ public void onItemClick(View view, View transitionSourceView,  int position) {
                 ActivityOptionsCompat options = ActivityOptionsCompat.makeSceneTransitionAnimation(getActivity(), transitionSourceView, "details");
                 startActivity(PetDetailsActivity.getStartIntent(getContext(), pets.get(position)), options.toBundle());
             }
+
+            @Override
+            public void onButtonClicked(ImageButton imageButton,PetFavoritesAdapter.ViewHolder viewHolder,int position ){
+                adapter.notifyItemRemoved(position);
```

Comment:
don't we need to remove the item at position from our datasource before we tell the adapter the item has been removed? Otherwise our data source is going to be out of sync from our view.